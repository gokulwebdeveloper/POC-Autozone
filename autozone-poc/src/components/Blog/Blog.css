.blog-wrapper .single-blog-area .hover-content p{
    height: 84px;
    line-height: 2;
    font-family: "Poppins", sans-serif;
    font-size: 14px;
}