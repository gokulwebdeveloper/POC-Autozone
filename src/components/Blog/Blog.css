
.blog-wrapper .single-blog-area{width:600px;}
.container-blog{ margin-right:35px; margin-left:50px;}

.blog-background-image{
    width: 100%;
    height: 390px;
    position: relative;
    z-index: 1;
    background-position: center center;
    background-size: cover;
    background-repeat: no-repeat;
    display: block; 
}

@media only screen and (max-width: 767px){
.blog-background-image {
    height: 250px;
}
.blog-wrapper .single-blog-area{width:fit-content;
  } 
.container-blog{margin:0px}
}

.blog-wrapper .single-blog-area .post-title p {
    display: block;
    font-size: 18px;
    font-weight: 600;
    color: black;
    line-height: 1.5;}
    @media only screen and (max-width: 767px) {
      .blog-wrapper .single-blog-area .post-title p {
        font-size: 14px; } }
    @media only screen and (min-width: 576px) and (max-width: 767px) {
      .blog-wrapper .single-blog-area .post-title p {
        font-size: 18px;} }

        .blog-wrapper .single-blog-area .hover-content .hover-post-title p {
            display: block;
            font-size: 18px;
            font-weight: 600;
            line-height: 1.5;
            color: black;
            margin-bottom: 20px; }
            @media only screen and (min-width: 992px) and (max-width: 1199px) {
              .blog-wrapper .single-blog-area .hover-content .hover-post-title p {
                font-size: 16px;
                margin-bottom: 10px; } }
            @media only screen and (max-width: 767px) {
              .blog-wrapper .single-blog-area .hover-content .hover-post-title p {
                font-size: 14px;
                margin-bottom: 10px; } }
            @media only screen and (min-width: 576px) and (max-width: 767px) {
              .blog-wrapper .single-blog-area .hover-content .hover-post-title p {
                font-size: 18px;
                margin-bottom: 20px; } }
               .blog-wrapper .single-blog-area .hover-content .hover-post-title p:hover {
              color: #0315ff; }

              .blog-wrapper .single-blog-area .hover-content .reading p{
                color: #0315ff;
                font-weight: 500; 
              }
 .row-blog{
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width:fit-content;
  
}
                 

