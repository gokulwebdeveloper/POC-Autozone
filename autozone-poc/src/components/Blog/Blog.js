import React, { useState, useEffect } from "react";
import "../Blog/Blog.css";
import { BrowserRouter as Router, NavLink, Link } from "react-router-dom";
import UseInfiniteScroll from "../../utlis/infiniteScroll/UseInfiniteScroll";
import background from "../../utlis/img/azo-blog-img/background_image.jpg";
import RetrieveJson from "../../utlis/RetrieveAPI/RetrieveJson";
import ScrollUpArrowBtn from "../../utlis/button/ScrollUpArrowBtn";

const Blog = (props) => {
  const [list, setList] = RetrieveJson("data/blog.json");

  const [listItems, setListItems] = useState(
    Array.from(Array(5).keys(), (n) => n + 1)
  );
  const [isFetching, setIsFetching] = UseInfiniteScroll(fetchMoreListItems);

  function fetchMoreListItems() {
    setTimeout(() => {
      setListItems((prevState) => [
        ...prevState,
        ...Array.from(Array(5).keys(), (n) => n + prevState.length + 1),
      ]);
      setIsFetching(false);
    }, 2000);
  }

  let sortedList = list.sort(
    (a, b) =>
      new Date(...a.recent_date.split("/").reverse()) -
      new Date(...b.recent_date.split("/").reverse())
  );
  let regularBlogList = sortedList.filter((item) => item.blogStatus);

  const bloglist = listItems.map((item) => {
    if (regularBlogList.length > item) {
      return (
        <div key={item}>
          <div className="col-12 col-lg-6">
            <div className="single-blog-area mb-50">
              <NavLink
                to="/singleblog"
                to={{
                  pathname: "/singleblog",
                  propsFor: {
                    img_id: regularBlogList[item].id,
                  },
                }}
              >
                <img src={regularBlogList[item].image} alt="" />
                <div className="post-title">
                  <a href="#">{regularBlogList[item].title}</a>
                </div>
                <div className="hover-content">
                  <div className="hover-post-title">
                    <a href="#">{regularBlogList[item].title}</a>
                  </div>
                  <p>{regularBlogList[item].content}</p>
                  <a href="#">
                    Continue reading <i className="fa fa-angle-right"></i>
                  </a>
                </div>
              </NavLink>
            </div>
          </div>
        </div>
      );
    }
  });

  return (
    <div>
      <ScrollUpArrowBtn />
      <div
        data-testid="background"
        className="blog-background-image"
        style={{ backgroundImage: "url(" + background + ")" }}
      ></div>
      <div className="blog-wrapper section-padding-80">
        <div className="container">
          <div className="row">{bloglist}</div>
          {isFetching && "Loading..."}
        </div>
      </div>
    </div>
  );
};

export default Blog;
