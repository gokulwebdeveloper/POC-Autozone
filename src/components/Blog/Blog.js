import React, { useState, useEffect, Fragment } from "react";
import "../Blog/Blog.css";
import { BrowserRouter as Router, NavLink, Link } from "react-router-dom";
import UseInfiniteScroll from "../../utlis/infiniteScroll/UseInfiniteScroll";
import RetrieveJson from "../../utlis/RetrieveAPI/RetrieveJson";
import ScrollUpArrowBtn from "../../utlis/button/ScrollUpArrowBtn";
import { Helmet } from "react-helmet";

const Blog = (props) => {
  const [list, setList] = RetrieveJson("data/blog.json");

  const [listItems, setListItems] = useState(
    Array.from(Array(5).keys(), (n) => n + 1)
  );
  const [isFetching, setIsFetching] = UseInfiniteScroll(fetchMoreListItems);

  function fetchMoreListItems() {
    setTimeout(() => {
      setListItems((prevState) => [
        ...prevState,
        ...Array.from(Array(5).keys(), (n) => n + prevState.length + 1),
      ]);
      setIsFetching(false);
    }, 2000);
  }

  let sortedList = list.sort(
    (a, b) =>
      new Date(...a.recent_date.split("/").reverse()) -
      new Date(...b.recent_date.split("/").reverse())
  );
  let regularBlogList = sortedList.filter((item) => item.blogStatus);

  const bloglist = listItems.map((item) => {
    if (regularBlogList.length > item) {
        const pathwithId = "/singleblog/"+regularBlogList[item].id;
      return (
        <div key={item}>
          <div className="col-12 col-lg-6">
            <div className="single-blog-area mb-50">
              <NavLink to={pathwithId} >
                <img src={regularBlogList[item].image} alt="" />
                <div className="post-title">
                  <p>{regularBlogList[item].title}</p>
                </div>
                <div className="hover-content">
                  <div className="hover-post-title">
                    <p>{regularBlogList[item].title}</p>
                  </div>
                  <p>{regularBlogList[item].content}</p>
                  <div className="reading">
                  <p>
                    Continue reading <i className="fa fa-angle-right"></i>
                  </p>
                  </div>
                </div>
              </NavLink>
            </div>
          </div>
        </div>
      );
    }
  });

  return (
    <Fragment>
      <Helmet>
        <meta charset="utf-8"/>
        <title>Autozone Blog</title>
        <meta name="description" content="This is Autozone blog page information"/>
      </Helmet>
      <ScrollUpArrowBtn />
      <div
        data-testid="background"
        className="blog-background-image"
        style={{ backgroundImage: "url(img/azo-blog-img/background_image.jpg)" }}
      ></div>
      <div className="blog-wrapper section-padding-80">
        <div className="container-blog">
          <div className="row-blog">{bloglist}</div>
          {isFetching && "Loading..."}
        </div>
      </div>
    </Fragment>
  );
};

export default Blog;
