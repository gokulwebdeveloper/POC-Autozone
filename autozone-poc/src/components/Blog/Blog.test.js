import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { shallow, mount } from "enzyme";
import Blog from "./Blog.js";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import blog1 from "../../utlis/img/bg-img/blog1.jpg";
import background from '../../utlis/img/azo-blog-img/background_image.jpg';


let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

describe("renders component with images", () => {
  const blog = shallow(<Blog />);

  it("renders an image", () => {
    //expect(blog.find("img").at(0).prop("src")).toBeTruthy();
    expect(blog.find("single-blog-area mb-50")).toBeTruthy();
  });

  it("SnapShot Match for blog screen", () => {
    const page = renderer.create(<Blog />).toJSON();
    expect(page).toMatchSnapshot();
  });

  it("has Id as background", () => {
    const blogPage = render(<Blog />);
    expect(blogPage.getByTestId("background")).toBeInTheDocument();
  });

  it("has backbground image with styles", () => {
    const blogPage = render(<Blog />);
    expect(blogPage.getByTestId("background")).toHaveStyle(
      "backgroundImage:"+background );
  });
});

describe("renders component with data", () => {
  const wrapper = shallow(<Blog />);

  const data = [
    {
      id: "1",
      image: blog1,
      content: "some data",
      description: "details",
      recent_date: "09/02/2020",
      blogStatus: true,
    },
    {
      id: "2",
      image: blog1,
      content: "some data",
      description: "details",
      recent_date: "19/05/2020",
      blogStatus: false,
    },
  ];
  it("renders without exploding", () => {
    mount(<Blog list={data} />);
  });

  it("renders with list title", () => {
    expect(wrapper.find("post-title")).toBeTruthy();
  });
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
