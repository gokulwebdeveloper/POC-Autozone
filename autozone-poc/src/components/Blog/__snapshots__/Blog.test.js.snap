// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component with images SnapShot Match for blog screen 1`] = `
<div>
  <div
    className="breadcumb_area breadcumb-style-two bg-img"
    data-testid="background"
    style={
      Object {
        "backgroundImage": "url(breadcumb2.jpg)",
      }
    }
  >
    <div
      className="container h-100"
    >
      <div
        className="row h-100 align-items-center"
      >
        <div
          className="col-12"
        >
          <div
            className="page-title text-center"
          >
            <h2>
              Fashion Blog
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="blog-wrapper section-padding-80"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-lg-6"
        >
          <div
            className="single-blog-area mb-50"
          >
            <img
              alt=""
              src="blog1.jpg"
            />
            <div
              className="post-title"
            >
              <a
                href="#"
              >
                Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
              </a>
            </div>
            <div
              className="hover-content"
            >
              <div
                className="hover-post-title"
              >
                <a
                  href="#"
                >
                  Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
                </a>
              </div>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce enim nulla, mollis eu metus in, sagittis fringilla tortor. Phasellus purus dignissim convallis.
              </p>
              <a
                href="#"
              >
                Continue reading 
                <i
                  className="fa fa-angle-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-lg-6"
        >
          <div
            className="single-blog-area mb-50"
          >
            <img
              alt=""
              src="blog2.jpg"
            />
            <div
              className="post-title"
            >
              <a
                href="#"
              >
                Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
              </a>
            </div>
            <div
              className="hover-content"
            >
              <div
                className="hover-post-title"
              >
                <a
                  href="#"
                >
                  Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
                </a>
              </div>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce enim nulla, mollis eu metus in, sagittis fringilla tortor. Phasellus purus dignissim convallis.
              </p>
              <a
                href="#"
              >
                Continue reading 
                <i
                  className="fa fa-angle-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-lg-6"
        >
          <div
            className="single-blog-area mb-50"
          >
            <img
              alt=""
              src="blog3.jpg"
            />
            <div
              className="post-title"
            >
              <a
                href="#"
              >
                Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
              </a>
            </div>
            <div
              className="hover-content"
            >
              <div
                className="hover-post-title"
              >
                <a
                  href="#"
                >
                  Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
                </a>
              </div>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce enim nulla, mollis eu metus in, sagittis fringilla tortor. Phasellus purus dignissim convallis.
              </p>
              <a
                href="#"
              >
                Continue reading 
                <i
                  className="fa fa-angle-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-lg-6"
        >
          <div
            className="single-blog-area mb-50"
          >
            <img
              alt=""
              src="blog4.jpg"
            />
            <div
              className="post-title"
            >
              <a
                href="#"
              >
                Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
              </a>
            </div>
            <div
              className="hover-content"
            >
              <div
                className="hover-post-title"
              >
                <a
                  href="#"
                >
                  Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
                </a>
              </div>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce enim nulla, mollis eu metus in, sagittis fringilla tortor. Phasellus purus dignissim convallis.
              </p>
              <a
                href="#"
              >
                Continue reading 
                <i
                  className="fa fa-angle-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-lg-6"
        >
          <div
            className="single-blog-area mb-50"
          >
            <img
              alt=""
              src="blog5.jpg"
            />
            <div
              className="post-title"
            >
              <a
                href="#"
              >
                Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
              </a>
            </div>
            <div
              className="hover-content"
            >
              <div
                className="hover-post-title"
              >
                <a
                  href="#"
                >
                  Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
                </a>
              </div>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce enim nulla, mollis eu metus in, sagittis fringilla tortor. Phasellus purus dignissim convallis.
              </p>
              <a
                href="#"
              >
                Continue reading 
                <i
                  className="fa fa-angle-right"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-lg-6"
        >
          <div
            className="single-blog-area mb-50"
          >
            <img
              alt=""
              src="blog6.jpg"
            />
            <div
              className="post-title"
            >
              <a
                href="#"
              >
                Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
              </a>
            </div>
            <div
              className="hover-content"
            >
              <div
                className="hover-post-title"
              >
                <a
                  href="#"
                >
                  Vivamus sed nunc in arcu cursus mollis quis et orci. Interdum et malesuada
                </a>
              </div>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce enim nulla, mollis eu metus in, sagittis fringilla tortor. Phasellus purus dignissim convallis.
              </p>
              <a
                href="#"
              >
                Continue reading 
                <i
                  className="fa fa-angle-right"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
